mlrun/serving/states.py

```diff
@@ -446,6 +450,36 @@ def set_flow(
     def supports_termination(self):
         return False
 
+    def verify_model_runner_step(self, step: "ModelRunnerStep"):
+        """
+        Verify ModelRunnerStep, can be part of Flow graph and models can not repeat in graph.
+        :param step: ModelRunnerStep to verify
+        """
+        if not isinstance(step, ModelRunnerStep):
+            return
+
+        root = self
+        while root.parent is not None:
+            root = self.parent
```

Comment:
```suggestion
            root = root.parent
```
Infinite loop otherwise... 😄

Reply:
good catch thanks